File: apps/ios/Sources/ConduitUI/Components/ConduitMark.swift
import SwiftUI

extension ConduitUI {
    /// The Conduit brand mark — the "terminal daemon": a rounded-square head
    /// with a cyan→green neon outline, top/bottom connector pills, `>` `<`
    /// squint eyes and a small smile. Vector reimplementation of the
    /// `ConduitMark` reference in the design handoff (BRAND.md §2), drawn on a
    /// 32×32 grid and scaled to `size`.
    ///
    /// - `color`: when set, the outline + pills render in this flat tint
    ///   (used for agent-tinted session avatars). When `nil`, the signature
    ///   cyan→green gradient is used.
    /// - `glow`: soft same-color outer glow (BRAND.md §3). On by default.
    struct ConduitMark: View {
        var size: CGFloat = 28
        var color: Color? = nil
        var glow: Bool = true

        // Theme-aware like the design `ConduitMark`: on a light canvas the pastel
        // cyan→green washes out and the near-white eyes vanish ("logo not
        // rendering well"), so light mode strokes the mark in the muted dark
        // accent and draws the eyes in the dark text colour. Glow is suppressed
        // on light (its cyan shadow read as a dark halo).
        @Environment(\.neonTheme) private var neon

        // BRAND.md §3 canonical tokens.
        private static let cyan = Color(hex: "#22d3ee")
        private static let green = Color(hex: "#3ef0a0")
        private static let eye = Color(hex: "#eafcff")

        var body: some View {
            let dark = neon.dark
            // Eyes/smile key off the *mode*, not the glow toggle: glow is an
            // independent switch that can be on in light mode, and the near-white
            // `eye` colour is invisible on a light canvas. Use dark text glyphs in
            // light mode (and suppress the cyan halo there — it reads as a dark blob).
            let eyeColor = dark ? Self.eye : neon.text
            let showGlow = glow && neon.glow && dark
            return Canvas { ctx, canvasSize in
                let s = canvasSize.width / 32.0
                func pt(_ x: CGFloat, _ y: CGFloat) -> CGPoint { CGPoint(x: x * s, y: y * s) }

                let outline: GraphicsContext.Shading = color.map { .color($0) }
                    ?? (dark
                        ? .linearGradient(
                            Gradient(colors: [Self.cyan, Self.green]),
                            startPoint: pt(4, 4), endPoint: pt(28, 28))
                        : .color(neon.accent))

                // Body — rounded square.
                let body = Path(
                    roundedRect: CGRect(x: 5.4 * s, y: 5.4 * s, width: 21.2 * s, height: 21.2 * s),
                    cornerRadius: 6.4 * s
                )
                ctx.stroke(body, with: outline, lineWidth: 2 * s)

                // Connector pills (cyan/green in dark, muted accent in light; flat tint when forced).
                let topFill: GraphicsContext.Shading = color.map { .color($0) } ?? .color(dark ? Self.cyan : neon.accent)
                let bottomFill: GraphicsContext.Shading = color.map { .color($0) } ?? .color(dark ? Self.green : neon.accent)
                ctx.fill(Path(roundedRect: CGRect(x: 14.4 * s, y: 4.4 * s, width: 3.2 * s, height: 2 * s), cornerRadius: 1 * s), with: topFill)
                ctx.fill(Path(roundedRect: CGRect(x: 14.4 * s, y: 25.6 * s, width: 3.2 * s, height: 2 * s), cornerRadius: 1 * s), with: bottomFill)

                // Face: `>` `<` squint eyes + smile.
                var face = Path()
                face.move(to: pt(11, 13.4)); face.addLine(to: pt(13.6, 15.4)); face.addLine(to: pt(11, 17.4))
                face.move(to: pt(21, 13.4)); face.addLine(to: pt(18.4, 15.4)); face.addLine(to: pt(21, 17.4))
                face.move(to: pt(13, 20)); face.addQuadCurve(to: pt(19, 20), control: pt(16, 22.4))
                ctx.stroke(face, with: .color(eyeColor),
                           style: StrokeStyle(lineWidth: 1.7 * s, lineCap: .round, lineJoin: .round))
            }
            .frame(width: size, height: size)
            .shadow(color: showGlow ? (color ?? Self.cyan).opacity(0.53) : .clear, radius: showGlow ? size * 0.1 : 0)
        }
    }
}

File: apps/android/app/src/main/kotlin/sh/nikhil/conduit/ui/ConduitMark.kt
package sh.nikhil.conduit.ui

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.CornerRadius
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.RoundRect
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.StrokeJoin
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.unit.Dp

// BRAND.md §3 canonical tokens.
private val MarkCyan = Color(0xFF22D3EE)
private val MarkGreen = Color(0xFF3EF0A0)
private val MarkEye = Color(0xFFEAFCFF)

/**
 * The Conduit brand mark — the "terminal daemon": a rounded-square head with a
 * cyan→green neon outline, top/bottom connector pills, `>` `<` squint eyes and
 * a small smile. Vector reimplementation of the design handoff `ConduitMark`
 * (BRAND.md §2), drawn on a 32×32 grid and scaled to [size]. Mirrors the iOS
 * `ConduitUI.ConduitMark`.
 *
 * @param color when non-null, the outline + pills render in this flat tint
 *   (agent-tinted avatars). When null, the signature cyan→green gradient.
 */
@Composable
fun ConduitMark(size: Dp, modifier: Modifier = Modifier, color: Color? = null) {
    val neon = LocalNeonTheme.current
    // Light theme: the signature cyan→green are pastel and wash out on a light
    // canvas, and the near-white eyes vanish entirely (device feedback: "logo
    // not rendering well"). Mirror the design `ConduitMark`, which is theme-aware
    // — on light it strokes the mark in the muted dark accent and draws the
    // `><` eyes in the dark text colour for contrast. Dark theme keeps the neon
    // cyan→green gradient + bright eyes.
    val markTop = if (color != null) color else if (neon.dark) MarkCyan else neon.accent
    val markBottom = if (color != null) color else if (neon.dark) MarkGreen else neon.accent
    // Eyes/smile key off the *mode*, not the glow toggle: glow is independent and
    // can be on in light mode, where the near-white eye colour vanishes. Use dark
    // text glyphs in light mode for contrast.
    val eyeColor = if (neon.dark) MarkEye else neon.text
    Canvas(modifier = modifier.size(size)) {
        val s = this.size.width / 32f
        fun p(x: Float, y: Float) = Offset(x * s, y * s)

        val outline = when {
            color != null -> SolidColor(color)
            neon.dark -> Brush.linearGradient(listOf(MarkCyan, MarkGreen), start = p(4f, 4f), end = p(28f, 28f))
            else -> SolidColor(neon.accent)
        }

        // Body — rounded square.
        val body = Path().apply {
            addRoundRect(
                RoundRect(
                    left = 5.4f * s, top = 5.4f * s, right = 26.6f * s, bottom = 26.6f * s,
                    cornerRadius = CornerRadius(6.4f * s),
                ),
            )
        }
        drawPath(body, brush = outline, style = Stroke(width = 2f * s))

        // Connector pills (top cyan, bottom green in dark; muted accent in light).
        drawRoundRect(color = markTop, topLeft = p(14.4f, 4.4f), size = Size(3.2f * s, 2f * s), cornerRadius = CornerRadius(1f * s))
        drawRoundRect(color = markBottom, topLeft = p(14.4f, 25.6f), size = Size(3.2f * s, 2f * s), cornerRadius = CornerRadius(1f * s))

        // Face: `>` `<` squint eyes + smile.
        val face = Path().apply {
            moveTo(11f * s, 13.4f * s); lineTo(13.6f * s, 15.4f * s); lineTo(11f * s, 17.4f * s)
            moveTo(21f * s, 13.4f * s); lineTo(18.4f * s, 15.4f * s); lineTo(21f * s, 17.4f * s)
            moveTo(13f * s, 20f * s); quadraticBezierTo(16f * s, 22.4f * s, 19f * s, 20f * s)
        }
        drawPath(face, color = eyeColor, style = Stroke(width = 1.7f * s, cap = StrokeCap.Round, join = StrokeJoin.Round))
    }
}
